### PR TITLE
Add CVE-2014-2120

### PR DESCRIPTION
**Pull request by PiranhaSa**
This is an old vulnerability where the hidden inputs are vulnerable to Rxss , the check is just to send the request with " and see if it's html encoded so if we found that it's reflected as it is , we can escape and we can get xss in the page . this is base on two pocs : https://packetstormsecurity.com/files/135813 and https://web.archive.org/web/20190327172646/https://www.trustwave.com/en-us/resources/security-resources/security-advisories/?fid=18914   . I couldn't find a vulnerable target till now all the ones from the google docks use html encoding 
image
still trying to find one

### CODE REVIEW

File: tests/exploits/cve_2014_2120_test.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        </body>
+    </html>
+    """
+
+
+def testCVE20142120_whenVulnerable_reportFinding(
+    requests_mock: req_mock.mocker.Mocker,
+) -> None:
+    """Unit test for CVE-2014-2120: case when target is vulnerable."""
+    # Mock the vulnerable response
```
Comment (rabsondev) [inline tests/exploits/cve_2014_2120_test.py:39]: I don't think this and other comments are necessary.

Comment (PiranhaSa) [reply tests/exploits/cve_2014_2120_test.py]: done

File: tests/exploits/cve_2014_2120_test.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    assert vulnerability.technical_detail == (
+        "http://localhost:80 is vulnerable to CVE-2014-2120, Cisco ASA WebVPN Login Page XSS Vulnerability"
+    )
+
+
+def testCVE20142120_whenSecure_reportNothing(
+    requests_mock: req_mock.mocker.Mocker,
+) -> None:
+    """Unit test for CVE-2014-2120: case when target is secure."""
+    # Mock the secure response
```
Comment (deadly-panda) [inline tests/exploits/cve_2014_2120_test.py:62]:
```suggestion
```

File: tests/exploits/cve_2014_2120_test.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+def create_mock_secure_response() -> str:
+    """Create a mock HTML response without the XSS vulnerability reflected."""
+    return """
+    <html>
+        <body>
+            <input type="hidden" name="auth_handle" value="test123&quot;">
+            <input type="hidden" name="username" value="&quot;test123">
+        </body>
+    </html>
+    """
```
Comment (elyousfi5) [inline tests/exploits/cve_2014_2120_test.py:31]: can be fixtures

File: tests/exploits/cve_2014_2120_test.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    requests_mock.get(
+        "http://localhost/+CSCOE+/logon.html?reason=2&a0=63&a1=&a2=&a3=0&next=&auth_handle=test123\"'&status=0&username=test123\"'&password_min=0&state=&tgroup=&serverType=0&password_days=0",
+        exc=requests.exceptions.Timeout,
+    )
+
+    exploit_instance = cve_2014_2120.CVE20142120XSSExploit()
+    target = definitions.Target("http", "localhost", 80)
+
+    vulnerabilities = exploit_instance.check(target)
+    assert len(vulnerabilities) == 0
```
Comment (elyousfi5) [inline tests/exploits/cve_2014_2120_test.py:105]:
```suggestion
    requests_mock.get(
        "http://localhost/+CSCOE+/logon.html?reason=2&a0=63&a1=&a2=&a3=0&next=&auth_handle=test123\"'&status=0&username=test123\"'&password_min=0&state=&tgroup=&serverType=0&password_days=0",
        exc=requests.exceptions.Timeout,
    )
    exploit_instance = cve_2014_2120.CVE20142120XSSExploit()
    target = definitions.Target("http", "localhost", 80)
    
    vulnerabilities = exploit_instance.check(target)
    
    assert len(vulnerabilities) == 0
```

Comment (nmasdoufi-ol) [reply tests/exploits/cve_2014_2120_test.py]: Done.

**Review by 3asm — APPROVED** [review]

**Pull request merged**